### PR TITLE
SDK-1204 Updated docs

### PR DESCRIPTION
**Pull request by sujaygarlanka**
### Goals :soccer:
- Update iterator code samples
- Update BoxError to BoxSDKError
- Update jazzy docs
- Update jazzy links

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sujaygarlanka has signed the CLA. Thanks for the pull request!

File: docs/usage/folders.md [hunk 1]
```diff
@@ -51,29 +51,31 @@ Get Folder Items
 
 To retrieve information about the items contained in a folder, call
 [`client.folders.listItems(folderId:usemarker:marker:offset:limit:sort:direction:fields:)`][get-folder-items]
-with the ID of the folder.  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the items in the folder.
+with the ID of the folder.  This method will return an iterator in the completion you can use to retrieve folder items.
```
Comment (PJSimon) [inline docs/usage/folders.md]:
```suggestion
with the ID of the folder.  This method will return an iterator in the completion, which is used to retrieve folder items.
```

File: docs/usage/folders.md [hunk 2]
```diff
@@ -148,6 +150,7 @@ client.folders.copy(
 
 [copy-folder]: http://opensource.box.com/box-ios-sdk/Classes/FoldersModule.html#/s:6BoxSDK13FoldersModuleC10copyFolder8folderId011destinationF2ID4name6fields10completionySS_S2SSgSaySSGSgys6ResultOyAA0F0CAA0A5ErrorOGctF
 
+NOTE: Is this an iterator or not?
```
Comment (PJSimon) [inline docs/usage/folders.md]: Is this a note for this code review?  Yes, the `listCollaborations` completion is:
`completion: @escaping Callback<PagingIterator<Collaboration>>`
Do you want to remove this comment?

File: docs/usage/groups.md [hunk 3]
```diff
@@ -202,20 +207,24 @@ client.groups.deleteMembership(membershipId: "12345") {
 Get Memberships for Group
 -------------------------
 
-To retrieve information about the memberships in a group, call [`client.groups.listMemberships(groupID: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-group] with the ID of the group to retrieve group memberships for.
+To retrieve information about the memberships in a group, call [`client.groups.listMemberships(groupID: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-group] with the ID of the group to retrieve group memberships for. This method will return an iterator object in the completion that you can use to the retrieve memberships.
```
Comment (PJSimon) [inline docs/usage/groups.md]:
```suggestion
To retrieve information about the memberships in a group, call [`client.groups.listMemberships(groupID: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-group] with the ID of the group to retrieve group memberships for. This method will return an iterator object in the completion, which is used to retrieve the memberships.
```

File: docs/usage/legal-holds.md [hunk 4]
```diff
@@ -132,21 +136,26 @@ Get Policy Assignments
 
 To retrieve legal hold policy assignments, call
 [`client.legalHolds.listPolicyAssignments(policyId: String, assignToType: String?, assignToId: String?, marker: String?, limit: String?, fields: [String]?)`][get-policy-assignments]
-with the ID of a policy.  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the policies assignments for a policy.
+with the ID of a policy.  This method will return an iterator object in the completion that you can use to retrieve policies assignments for a policy.
```
Comment (PJSimon) [inline docs/usage/legal-holds.md]:
```suggestion
with the ID of a policy.  This method will return an iterator object in the completion, which is used to retrieve policy assignments for a policy.
```

File: docs/usage/legal-holds.md [hunk 5]
```diff
@@ -41,20 +41,24 @@ Get Legal Hold Policies
 
 To retrieve information about the items contained in a folder, call
 [`client.legalHolds.listForEnterprise(policyName: String, marker: String?, limit: Int?, fields: [String]?)`][get-legal-hold-policies]
-with the ID of the policy.  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the policies in the enterprise.
+with the ID of the policy.  This method will return an iterator object in the completion you can use to retrieve policies in the enterprise.
```
Comment (PJSimon) [inline docs/usage/legal-holds.md]:
```suggestion
with the ID of the policy.  This method will return an iterator object in the completion, which is used to retrieve policies in the enterprise.
```

File: docs/usage/groups.md [hunk 6]
```diff
@@ -249,21 +262,25 @@ membershipIterator.getNextItems() {
 Get Collaborations for Group
 ----------------------------
 
-To retrieve all group collaborations for a given group, call [`client.groups.listCollaborations(groupId: String, offset: Int?, limit: Int? fields: [String]?)`][get-collaborations-for-group] with the ID of the group to retrieve collaborations for.
+To retrieve all group collaborations for a given group, call [`client.groups.listCollaborations(groupId: String, offset: Int?, limit: Int? fields: [String]?)`][get-collaborations-for-group] with the ID of the group to retrieve collaborations for. This method will return an iterator object in the completion that you can use to retrieve the collaborations.
```
Comment (PJSimon) [inline docs/usage/groups.md]:
```suggestion
To retrieve all group collaborations for a given group, call [`client.groups.listCollaborations(groupId: String, offset: Int?, limit: Int? fields: [String]?)`][get-collaborations-for-group] with the ID of the group to retrieve collaborations for. This method will return an iterator object in the completion, which is used to retrieve the collaborations.
```

File: docs/usage/groups.md [hunk 7]
```diff
@@ -226,20 +235,24 @@ Get Memberships for User
 ------------------------
 
 To retrieve information about the group memberships for a given user, call
-[`client.groups.listMembershipsForUser(userId: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-user] with the ID of the user to retreive group memberships for.
+[`client.groups.listMembershipsForUser(userId: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-user] with the ID of the user to retreive group memberships for. This method will return an iterator object in the completion that you can use to retrieve the memberships.
```
Comment (PJSimon) [inline docs/usage/groups.md]:
```suggestion
[`client.groups.listMembershipsForUser(userId: String, offset: Int?, limit: Int?, fields: [String]?)`][get-memberships-for-user] with the ID of the user to retreive group memberships for. This method will return an iterator object in the completion, which is used to retrieve the memberships.
```

File: docs/usage/groups.md [hunk 8]
```diff
@@ -84,19 +84,24 @@ client.groups.get(groupId: "12345") {
 Get Enterprise Groups
 ---------------------
 
-To retrieve information about the groups within the enterprise, call [`client.groups.listForEnterprise(name: String?, offset: Int?, limit: Int?, fields: [String]?)`][get-enterprise-groups]. You can also pass in a `name` paramter to act as a filter. This method will return an iterator object you can use to retrieve successive pages of result, where each page contains some of the groups in the enterprise.
+To retrieve information about the groups within the enterprise, call [`client.groups.listForEnterprise(name: String?, offset: Int?, limit: Int?, fields: [String]?)`][get-enterprise-groups]. You can also pass in a `name` paramter to act as a filter. This method will return an iterator object in the completion that you can use to retrieve groups in the enterprise.
```
Comment (PJSimon) [inline docs/usage/groups.md]:
```suggestion
To retrieve information about the groups within the enterprise, call [`client.groups.listForEnterprise(name: String?, offset: Int?, limit: Int?, fields: [String]?)`][get-enterprise-groups]. You can also pass in a `name` paramter to act as a filter. This method will return an iterator object in the completion, which is used to retrieve groups in the enterprise.
```

File: docs/usage/legal-holds.md [hunk 9]
```diff
@@ -207,20 +216,24 @@ Get File Version Legal Holds
 
 To retrieve all of the non-deleted legal holds for a single legal hold policy, call
 [`client.legalHolds.listFileVersionPolicies(policyId: String, marker: String?, limit: String?, fields: [String]?)`][get-file-version-legal-holds]
-with the ID of a policy.  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the non-deleted legal holds for a policy.
+with the ID of a policy.  This method will return an iterator object in the completion that you can use to retrieve legal holds for a policy.
```
Comment (PJSimon) [inline docs/usage/legal-holds.md]:
```suggestion
with the ID of a policy.  This method will return an iterator object in the completion, which is used to retrieve legal holds for a policy.
```

File: docs/usage/metadata.md [hunk 10]
```diff
@@ -181,19 +181,24 @@ List Metadata Templates
 
 To retrieve the collection of available metadata templates in a particular scope, call
 [`client.metadata.listEnterpriseTemplates(scope:marker:limit:)`][list-templates]
-with the scope.
+with the scope. This method will return an iterator object in the completion that you can use to retrieve metadata templates.
```
Comment (PJSimon) [inline docs/usage/metadata.md]:
```suggestion
with the scope. This method will return an iterator object in the completion, which is used to retrieve metadata templates.
```

File: docs/usage/recent-items.md [hunk 11]
```diff
@@ -15,23 +15,24 @@ Get Recent Items
 ----------------
 
 To get recently accessed items, call
-[`client.recentItems.list(marker: String?, limit: Int?, fields: [String]?)`][get-recent-items]. This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the recent items.
+[`client.recentItems.list(marker: String?, limit: Int?, fields: [String]?)`][get-recent-items]. This method will return an iterator object in the completion you can use to retrieve recent items.
```
Comment (PJSimon) [inline docs/usage/recent-items.md]:
```suggestion
[`client.recentItems.list(marker: String?, limit: Int?, fields: [String]?)`][get-recent-items]. This method will return an iterator object in the completion, which is used to retrieve recent items.
```

File: docs/usage/search.md [hunk 12]
```diff
@@ -15,26 +15,31 @@ Content Search
 
 To get a list of items matching a search query, call [`client.search.query(query:...)`][search] with the
 string to query for.  There are many possible options for advanced search filtering, which can be used to narrow down
-the search results.
+the search results. This method will return an iterator object in the completion you can use to get the results.
```
Comment (PJSimon) [inline docs/usage/search.md]:
```suggestion
the search results. This method will return an iterator object in the completion, which is used to get the results.
```

File: docs/usage/search.md [hunk 13]
```diff
@@ -46,29 +51,34 @@ Metadata Search
 
 To search within metadata for specific values, use the helper methods included in the SDK to construct the metadata
 query and call [`client.search.query(query:...)`][search] with the metadata filters and optionally a
-search query.  If no additional content query string is needed, just set the `query` parameter to `nil`.
+search query.  If no additional content query string is needed, just set the `query` parameter to `nil`. This method will return an iterator object in the completion you can use to get the results.
```
Comment (PJSimon) [inline docs/usage/search.md]:
```suggestion
search query.  If no additional content query string is needed, just set the `query` parameter to `nil`. This method will return an iterator object in the completion, which is used to get the results. 
```

File: docs/usage/storage-policies.md [hunk 14]
```diff
@@ -38,20 +38,24 @@ Get Storage Policies
 --------------------
 
 To retrieve the storage policies in an enterprise, call
-[`client.storagePolicies.list(marker: String?, limit: Int?, fields: [String]?)`][get-storage-policies].  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the policies in the enterprise.
+[`client.storagePolicies.list(marker: String?, limit: Int?, fields: [String]?)`][get-storage-policies].  This method will return an iterator in the completion that can be used to get the policies.
```
Comment (PJSimon) [inline docs/usage/storage-policies.md]:
```suggestion
[`client.storagePolicies.list(marker: String?, limit: Int?, fields: [String]?)`][get-storage-policies].  This method will return an iterator in the completion, which is used to get the policies.
```

File: docs/usage/users.md [hunk 15]
```diff
@@ -171,18 +171,24 @@ Get Enterprise Users
 
 To retrieve the users in an enterprise, call
 [`client.users.listForEnterprise(filterTerm:fields:offset:limit:)`][get-enterprise-users].
-The method returns an iterator object that can be used to page through the entire collection of users.
+The method returns an iterator object in the completion that you can use to get the users.
```
Comment (PJSimon) [inline docs/usage/users.md]:
```suggestion
The method returns an iterator object in the completion, which is used to get the users.
```

File: docs/usage/webhooks.md [hunk 16]
```diff
@@ -38,22 +38,24 @@ Get Webhooks
 ----------------
 
 To retrieve information about webhooks in an enterprise, call
-[`client.webhook.list(webhookId: String, marker: String?, limit: Int?, fields: [String]?)`].  This method will return an iterator object you can use to retrieve successive pages of
-results, where each page contains some of the webhooks in the enterprise.
+[`client.webhook.list(webhookId: String, marker: String?, limit: Int?, fields: [String]?)`].  This method will return an iterator object in the completion you can use to get the webhooks.
```
Comment (PJSimon) [inline docs/usage/webhooks.md]:
```suggestion
[`client.webhook.list(webhookId: String, marker: String?, limit: Int?, fields: [String]?)`].  This method will return an iterator object in the completion, which is used to get the webhooks.
```

File: docs/usage/files.md [hunk 17]
```diff
@@ -322,19 +331,24 @@ Get File Tasks
 --------------
 
 To retrieve a list of file tasks, call [`client.files.listTasks(forFile:fields:)`][get-tasks] with the ID of the
-file.  This method returns an iterator that can be used to page through the collection of file comments.
+file.  This method returns an iterator in the completion that can be used to retrieve file comments.
```
Comment (PJSimon) [inline docs/usage/files.md]:
```suggestion
file.  This method returns an iterator in the completion, which is used to retrieve file comments.
```

File: docs/usage/files.md [hunk 18]
```diff
@@ -298,20 +302,25 @@ Get File Comments
 
 To retrieve a list of comments on the file, call
 [`client.files.listComments(forFile:offset:limit:fields:)`][get-comments]
-with the ID of the file.  This method returns an iterator that can be used to page through the collection
+with the ID of the file.  This method returns an iterator in the completion that can be used to page through the collection
```
Comment (PJSimon) [inline docs/usage/files.md]:
```suggestion
with the ID of the file.  This method returns an iterator in the completion, which is used to page through the collection
```

File: docs/usage/files.md [hunk 19]
```diff
@@ -273,20 +273,24 @@ Get File Collaborations
 
 To retrieve a list of collaborations on a file, call
 [`client.files.listCollaborations(forFile:marker:limit:fields:)`][get-collaborations]
-with the ID of the file.  This method returns an iterator that can be used to page through the collection
-of file collaborations.
+with the ID of the file.  This method returns an iterator in the completion that can be used to retrieve file collaborations.
```
Comment (PJSimon) [inline docs/usage/files.md]:
```suggestion
with the ID of the file.  This method returns an iterator in the completion, which is used to retrieve file collaborations.
```

File: docs/usage/devicepins.md [hunk 20]
```diff
@@ -35,19 +35,24 @@ client.devicePins.get(devicePinId: "11111", fields: ["product_name"]) { (result:
 Get Device Pins for Enterprise
 ------------------------------
 
-To retrieve information about the device pins active for the enterprise, call [`client.devicePins.listForEnterprise(enterpriseId: String, marker: String?, limit: Int?, direction: OrdeDirection?, fields: [String]?)`][get-device-pins] with the ID of the enterpise. This method will return an iterator object you can use to retrieve successive pages of results, where each page contains some of the device pins for the enterprise.
+To retrieve information about the device pins active for the enterprise, call [`client.devicePins.listForEnterprise(enterpriseId: String, marker: String?, limit: Int?, direction: OrdeDirection?, fields: [String]?)`][get-device-pins] with the ID of the enterpise. This method will return an iterator object in the completion that you can use to retrieve device pins for the enterprise.
```
Comment (PJSimon) [inline docs/usage/devicepins.md]:
```suggestion
To retrieve information about the device pins active for the enterprise, call [`client.devicePins.listForEnterprise(enterpriseId: String, marker: String?, limit: Int?, direction: OrdeDirection?, fields: [String]?)`][get-device-pins] with the ID of the enterpise. This method will return an iterator object in the completion, which is used to retrieve device pins for the enterprise.
```

File: docs/usage/collaborations.md [hunk 21]
```diff
@@ -126,17 +126,25 @@ Get Pending Collaborations
 
 To retrieve a list of the pending collaborations requiring the user to accept or reject them, call
 [`client.collaborations.listPendingForEnterprise(offset:limit:fields:)`][get-pending-collaborations].
-The method returns an iterator that can be used to page through the collection of pending collaborations.
+The method returns an iterator in the completion that can be used to get pending collaborations.
```
Comment (PJSimon) [inline docs/usage/collaborations.md]:
```suggestion
The method returns an iterator in the completion, which is used to get pending collaborations.
```

**Review by PJSimon — APPROVED** [review]

**Review by PJSimon — APPROVED** [review]

**Pull request merged**